Corzbank.Web/ClientApp/src/app/header/wallet/wallet.component.html

```diff
@@ -50,9 +48,112 @@ <h2>Deposits</h2>
+					(click)="selectDeposit(deposit.id)">
+					<div class="deposit-title">
+						<span>{{ deposit.amount }}$</span>
+						<span>Duration: {{ deposit.duration }}m</span>
+					</div>
+					<h4>End date: {{ deposit.endDate | date: "dd-MM-yyyy" }}</h4>
+				</div>
+			</div>
+			<div class="submit-button" *ngIf="cards.length < 3">
+				<button (click)="openDepositMenu()">Open Deposit</button>
+			</div>
+		</div>
 	</div>
-	<div *ngIf="cardsIsDisplayed && cardDataIsDisplayed">
+	<div class="open-card-menu-block" *ngIf="openCardMenuIsDisplayed && cardsIsDisplayed">
+		<div class="title-open-card-menu title-transactions">
+			<h2>Open Card</h2>
+		</div>
+		<div class="open-card-menu" [formGroup]="createCardForm">
+			<div class="field">
+				<div class="card-type">
+					<span>Card Type:</span>
+					<select formControlName="cardType">
+						<option *ngFor="let cardType of cardTypeKeys" [value]="cardType">
+							{{ CardType[cardType] }}
+						</option>
+					</select>
+				</div>
+				<small *ngIf="cardType && cardType.invalid && cardType.touched">Card Type is Required</small>
+			</div>
+			<div class="field">
+				<div class="payment-system">
+					<span>Payment System:</span>
+					<select formControlName="paymentSystem">
+						<option *ngFor="let paymentSystem of paymentSystemKeys" [value]="paymentSystem">
+							{{ PaymentSystem[paymentSystem] }}
+						</option>
+					</select>
+				</div>
+				<small *ngIf="
+						paymentSystem &&
+						paymentSystem.invalid &&
+						paymentSystem.touched
+					">Payment System is Required</small>
+			</div>
+			<div class="field">
+				<div class="secret-word">
+					<span>Secret Word:</span>
+					<input type="text" formControlName="secretWord" />
+				</div>
+				<small *ngIf="secretWord && secretWord.invalid && secretWord.touched">Secret Word is Required</small>
+			</div>
+			<div class="open-card submit-button">
+				<button (click)="createCard()" [disabled]="createCardForm.invalid">
+					Create Card
+				</button>
+			</div>
+		</div>
+	</div>
+	<div class="open-deposit-menu-block" *ngIf="openDepositMenuIsDisplayed && depositsIsDisplayed">
+		<div class="title-open-deposit-menu title-transactions">
+			<h2>Open Deposit</h2>
+		</div>
+		<div class="open-deposit-menu">
+			<div class="field">
+				<div class="card">
+					<span>Select Card: </span>
+					<select [(ngModel)]="cardForDeposit.id">
+						<option *ngFor="let card of cards" [value]="card.id">
+							{{ card.cardNumber }}
+						</option>
+					</select>
+				</div>
+				<small *ngIf="cardForDeposit.balance < amountOfDeposit">Balance is not enough</small>
+			</div>
+			<div class="field">
+				<div class="amount">
+					<span>Amount:</span>
+					<input type="text" [(ngModel)]="amountOfDeposit" min="1" maxlength="7" />
+				</div>
+			</div>
+			<div class="duration">
+				<label>
+					<button (click)="decrement()">-</button>
+					Duration: {{ durationOfDeposit }}m
```

Comment:
What do you think of depositDuration name?

Reply:
resolved